server/src/main/java/org/candlepin/servlet/filter/EventFilter.java

```diff
@@ -47,29 +49,44 @@
 
     private static Logger log = LoggerFactory.getLogger(EventFilter.class);
 
+    private I18n i18n;
     private Injector injector;
 
+
     @Inject
-    public EventFilter(Injector injector) {
+    public EventFilter(Injector injector, I18n i18n) {
         this.injector = injector;
+        this.i18n = i18n;
     }
 
     @Override
     public void doFilter(ServletRequest request, ServletResponse response,
         FilterChain chain) throws IOException, ServletException {
-        // EventSink must get injected here instead of in the constructor
-        // because on creation of the filter we will be out of the
-        // CandlepinRequestScope as the filter must be a singleton.
-        EventSink eventSink = injector.getInstance(EventSink.class);
-        TeeHttpServletResponse resp = new TeeHttpServletResponse((HttpServletResponse) response);
-        chain.doFilter(request, resp);
-        Status status = Status.fromStatusCode(resp.getStatus());
-        if (status.getFamily() == Status.Family.SUCCESSFUL) {
-            eventSink.sendEvents();
+
+        String contentType = request.getContentType();
+        try {
+            // To validate the media type by parsing the supplied content type.
+            // If not valid, it throws an IllegalArgumentException.
+            // If valid, the below line of code works as expected.
+            MediaType type = contentType == null ? null : MediaType.valueOf(contentType);
+
+            // EventSink must get injected here instead of in the constructor
+            // because on creation of the filter we will be out of the
+            // CandlepinRequestScope as the filter must be a singleton.
+            EventSink eventSink = injector.getInstance(EventSink.class);
+            TeeHttpServletResponse resp = new TeeHttpServletResponse((HttpServletResponse) response);
+            chain.doFilter(request, resp);
+            Status status = Status.fromStatusCode(resp.getStatus());
+            if (status.getFamily() == Status.Family.SUCCESSFUL) {
+                eventSink.sendEvents();
+            }
+            else {
+                eventSink.rollback();
+                log.debug("Request failed, skipping event sending, status={}", status);
+            }
         }
-        else {
-            eventSink.rollback();
-            log.debug("Request failed, skipping event sending, status={}", status);
+        catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(this.i18n.tr("Invalid Content-Type {0}", contentType));
```

Comment:
Unfortunately this solution will lead to the same stacktrace but with a `IllegalArgumentException` instead of an NPE. Because we're not at the resteasy layer here, we can't rely on the `ExceptionMappers` to send back a proper response, and instead we have to build/send one ourselves. The following piece of code at the start of the method should work, and also avoids the whole Tomcat HTML output:
```
    @Override
    public void doFilter(ServletRequest request, ServletResponse response,
        FilterChain chain) throws IOException, ServletException {

        I18n i18n = injector.getInstance(I18n.class);
        // Validate media type contains a slash, to avoid RESTEasy's inability to handle this
        String contentType = request.getContentType();
        if(contentType != null && !contentType.contains("/")) {
            HttpServletResponse httpResponse = (HttpServletResponse) response;
            httpResponse.setContentType("text/plain");
            httpResponse.setCharacterEncoding("UTF-8");
            httpResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST);
            httpResponse.getWriter().write(i18n.tr("Invalid Content-Type {0}", contentType));
            return;
        }
```

Reply:
Updated the code in the Filter class.